Using-the-Compiler-API.md

```diff
@@ -607,6 +607,42 @@ const result = printer.printNode(ts.EmitHint.Unspecified, makeFactorialFunction(
 console.log(result);
 ```
 
+## Type Checker APIs
+
+Programs contain a `TypeChecker` object that provides methods for retrieving and reasoning about the types of AST nodes.
+Type checker APIs generally work with two types of representations alongside AST nodes:
+
+* Symbol: describes how the type system views a declared entity, such as a class, function, or variable.
+* Type: describes the backing type that entities may be declared as.
+  These often have a backing Symbol pointing to their declaration(s).
+
+For example, given a single function declaration like:
+
+```ts
+function greet() {
+  console.log("Hello, world!");
+}
+```
+
+TypeScript would create the following representations:
+
+* Symbol (`Symbol(greet)`): describing the `greet` function declaration.
+  It includes the function's name and a list of declarations (here, just one).
+* Type (`() => void`): describing the type of `greet`, a function with no parameters and a `void` return.
+  Its backing symbol points to `Symbol(greet)`.
```

Comment:
I dunno if this is actually better or worse - let me know what you think.

```suggestion
TypeScript will create a `Symbol` in the containing scope for `greet`.
When looking up ("resolving") an identifier with the name `greet`, that `Symbol` can be retrieved.
This `Symbol` will contain information about how `greet` was declared, and can be used to gain information about the type of `greet`.

On that note, TypeScript will also create a type describing `greet`, which is effectively the type `() => void` - a function with no parameters and a `void` return type.
In this case, the type will be backed by the original symbol associated with `greet`.
```

Reply:
Agreed, this is more clear - thanks!